src/Features/VisualBasic/Portable/CodeRefactorings/AddNamedArguments/VisualBasicAddNamedArgumentsCodeRefactoringProvider.vb

```diff
@@ -0,0 +1,143 @@
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.AddNamedArguments
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(VisualBasicAddNamedArgumentsCodeRefactoringProvider)), [Shared]>
+    Friend Class VisualBasicAddNamedArgumentsCodeRefactoringProvider
+        Inherits AbstractAddNamedArgumentsCodeRefactoringProvider(Of ArgumentSyntax)
+
+        Protected Overrides Function GetNamedArguments(arguments As SeparatedSyntaxList(Of ArgumentSyntax), parameters As ImmutableArray(Of IParameterSymbol), ByRef hasLiteral As Boolean) As ArgumentSyntax()
+            ' An argument list with omitted arguments may or may not be already fixed
+            Dim alreadyFixed = True
+            Dim result = ArrayBuilder(Of ArgumentSyntax).GetInstance()
+            For index = 0 To arguments.Count - 1
+                Dim argument = arguments(index)
+                Dim parameter = parameters(index)
+                If argument.IsNamed Then
+                    result.Add(argument)
+                ElseIf argument.IsOmitted Then
+                    Continue For
+                ElseIf parameter.IsParams Then
+                    ' Named arguments cannot match a ParamArray parameter
+                    result.Free()
+                    Return Nothing
+                Else
+                    alreadyFixed = False
+                    If Not hasLiteral AndAlso IsLiteral(argument) Then
+                        hasLiteral = True
+                    End If
+                    Dim simpleArgument = DirectCast(argument, SimpleArgumentSyntax)
+                    result.Add(simpleArgument.WithNameColonEquals(NameColonEquals(IdentifierName(parameter.Name))).WithTriviaFrom(simpleArgument.Expression))
+                End If
+            Next
+            If alreadyFixed Then
+                result.Free()
+                Return Nothing
+            Else
+                Return result.ToArrayAndFree()
+            End If
+        End Function
+
+        Protected Overrides Function GetTargetNode(node As SyntaxNode) As SyntaxNode
+            If node.IsKind(SyntaxKind.ConditionalAccessExpression) Then
+                Return DirectCast(node, ConditionalAccessExpressionSyntax).WhenNotNull
+            End If
+            Return node
+        End Function
+
+        Protected Overrides Function IsCandidate(node As SyntaxNode) As Boolean
+            Select Case node.Kind()
+                Case SyntaxKind.InvocationExpression
+                    ' If this is an indexer which is inside a conditional access, go up the tree
+                    Dim invocation = DirectCast(node, InvocationExpressionSyntax)
+                    Return invocation.Expression IsNot Nothing
+
+                Case SyntaxKind.ConditionalAccessExpression
+                    ' If this is a conditional invocation, pass
```

Comment:
This means that we don't suggest code fix for `Method` in  `expr?.Method(1)` on `expr`.